lang/py/test/test_schema.py

```diff
@@ -408,15 +408,14 @@ def test_parse(self):
     for example in EXAMPLES:
       try:
         schema.parse(example.schema_string)
+      except (schema.AvroException, schema.SchemaParseException):
```

Comment:
@Fokko I could use some guidance here -- I believe this test is more correct than it was in the previous implementation, but it is now failing on a test case with schema

```
{"type": "long", "logicalType": "date"}
```

But the implementation allows this to pass [on line 921](https://github.com/apache/avro/blob/master/lang/py/src/avro/schema.py#L921) where it falls back to the primitive type if the logical type doesn't match any earlier condition.

Should we raise an exception whenever a logicalType is specified if it doesn't match the expected primitive?

Reply:
I think we should raise an exception, because on one side, the [spec|http://avro.apache.org/docs/current/spec.html#Date] clearly specifies the required physical type, on the other side, Java validates logical types [here|https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java#L316], so better to stay consistent across languages.

Reply:
Thanks, @nandorKollar. I've opened AVRO-2580 for this and will address that issue first to avoid mixing contexts in this PR.

Reply:
Sounds good to me, thanks!

Reply:
That change is in #668. Please take a look if you get the chance.